### PR TITLE
Cleaned up fake emails for anonymous and system users 

### LINKED ISSUE #20095: E-mail address used for system/anonymous users in dotCMS are confusing/unclear
**Is your feature request related to a problem? Please describe.**
The e-mails used for system actions (system@fakedotcmsemail.org) and anonymous users (anonymous@fakedotcmsemail.org) are confusing and can cause security questions when seen.

Ticket: https://dotcms.zendesk.com/agent/tickets/103780

**Describe the solution you'd like**
Change the e-mail addresses to use language that doesn't insinuate they are not real.

### PR DESCRIPTION
**Pull request by swicken-dotcms**
Cleaned up fake emails for anonymous and system users to be more friendly. Updated same emails in tests to reflect this change.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java [hunk 1]
```diff
@@ -231,15 +231,15 @@ private synchronized User _getAnonymousUser() throws DotDataException {
         try {
             user = userFactory.loadUserById(CMS_ANON_USER_ID);
         } catch (DotDataException e) {
-            user = createUser(CMS_ANON_USER_ID, "anonymous@dotcmsfakeemail.org");
+            user = createUser(CMS_ANON_USER_ID, "anonymous@dotcms.anonymoususer");
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java:234]: We should use a constant here

File: dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java [hunk 2]
```diff
@@ -231,15 +231,15 @@ private synchronized User _getAnonymousUser() throws DotDataException {
...
             user.setUserId(CMS_ANON_USER_ID);
             user.setFirstName("Anonymous");
             user.setLastName("User");
             user.setCreateDate(new java.util.Date());
             user.setCompanyId(PublicCompanyFactory.getDefaultCompanyId());
             userFactory.saveUser(user);
         } catch (NoSuchUserException e) {
-            user = createUser(CMS_ANON_USER_ID, "anonymous@dotcmsfakeemail.org");
+            user = createUser(CMS_ANON_USER_ID, "anonymous@dotcms.anonymoususer");
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java:242]: same here

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java (hunk 2 above)
Comment (swicken-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java]: Agreed, moved to constant

File: dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java (hunk 1 above)
Comment (swicken-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java]: Agreed, moved to constant

**Review by dsilvam — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**